packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -66,6 +88,31 @@ async function* loadAllSequencedMessages(
     let requests = 0;
     let opsStorage = 0;
 
+    // reading only 1 op to test if there is mismatch
+    const teststream = deltaStorage.fetchMessages(
+        lastSeq + 1,
+        lastSeq + 2);
+
+    let statusCode;
+    let innerMostErrorCode;
+    let response;
+
+    try {
+        await teststream.read();
+    } catch (error) {
+        statusCode = error.getTelemetryProperties().statusCode;
+        innerMostErrorCode = error.getTelemetryProperties().innerMostErrorCode;
+        // if there is gap between ops, catch the error and check it is the error we need
+        if (statusCode !== 410 || innerMostErrorCode !== "fluidDeltaDataNotAvailable") {
+            throw error;
+        }
+        // get firstAvailableDelta from the error response, and set current sequence number to that
+        response = JSON.parse(error.getTelemetryProperties().response);
+        lastSeq = response.error.firstAvailableDelta - 1;
+        currSeq = lastSeq + 1;
```

Comment:
Just set it to "firstAvailableDelta" after renaming as per other comment instead of last + 1.